DuckDuckGo/Onboarding/ContextualOnboarding/OnboardingFireButtonDialogViewModel.swift

```diff
@@ -17,14 +17,27 @@
 //
 
 import Foundation
+import Combine
 
-public struct OnboardingFireButtonDialogViewModel {
-    let onDismiss: () -> Void
+public class OnboardingFireButtonDialogViewModel: ObservableObject {
 
-    func skip() {
+    var onDismiss: () -> Void
+    var onGotItPressed: () -> Void
+
+    init(onDismiss: @escaping () -> Void, onGotItPressed: @escaping () -> Void) {
+        self.onDismiss = onDismiss
+        self.onGotItPressed = onGotItPressed
+    }
+
+    func highFive() {
+        onGotItPressed()
         onDismiss()
     }
 
+    func skip() {
+        onGotItPressed()
+    }
+
     @MainActor
     func tryFireButton() {
         FireCoordinator.fireButtonAction()
```

Comment:
if we want to test `tryFireButton`  we could probably inject `fireButtonAction` function